connection/cluster/ClusterClient.java

```diff
@@ -210,11 +217,12 @@ abstract class FailsafeTask<RESULT> {
         private final int PRIMARY_REPLICA_TASK_MAX_RETRIES = 10;
         private final int FETCH_REPLICAS_MAX_RETRIES = 10;
         private final int WAIT_FOR_PRIMARY_REPLICA_SELECTION_MS = 2000;
-
         private final String database;
+        private int roundRobinIndex;
```

Comment:
This should not be per `FailsafeTask`. As it stands, we always begin by querying the first address in the list.

Reply:
After discussion, we decided to leave this here so a single task's `fetchDatabaseReplicas` retries are guaranteed to go roundRobin without interfering with each other.

Reply:
To clarify, the existence of a preferred replica makes this a non-issue.